### PR TITLE
ARM ID validation

### PR DESCRIPTION
**Pull request by matthchr**
Related to #765

### CODE REVIEW

**Review by raych1 — CHANGES_REQUESTED** [review]
Can you add some unit tests for arm-id validation at
https://github.com/Azure/oav/blob/develop/test/modelValidatorTests.ts

**Comment by matthchr** [comment]
I've added a unit test @raych1. Let me know if you feel there should be more. I tried to follow the other format tests that I saw (`byte`).

**Review by sanjaiganesh — APPROVED** [review]

**Review by raych1 — APPROVED** [review]

File: lib/swaggerValidator/modelValidator.ts [hunk 1]
```diff
@@ -573,6 +573,16 @@ export class SwaggerExampleValidator {
             }
           }
         }
+
+        if (
+          (err.code as any) === "MISSING_RESOURCE_ID" &&
+          exampleContent.responses[`${statusCode}`].body &&
```
Comment (jianyexi) [inline lib/swaggerValidator/modelValidator.ts]:
```suggestion
          exampleContent.responses[statusCode].body &&
```

File: lib/swaggerValidator/modelValidator.ts [hunk 2]
```diff
@@ -573,6 +573,16 @@ export class SwaggerExampleValidator {
...
+          Object.keys(exampleContent.responses[`${statusCode}`].body).length === 0
```
Comment (jianyexi) [inline lib/swaggerValidator/modelValidator.ts]:
```suggestion
          Object.keys(exampleContent.responses[statusCode].body).length === 0
```

**Review by jianyexi — APPROVED** [review]

**Pull request merged**